toolchain/check/modifiers_allowed.h

```diff
@@ -0,0 +1,40 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_CHECK_MODIFIERS_ALLOWED_H_
+#define CARBON_TOOLCHAIN_CHECK_MODIFIERS_ALLOWED_H_
+
+#include "toolchain/check/context.h"
+
+namespace Carbon::Check {
+
+// Reports a diagnostic (using `decl_name`) if access control modifiers on this
+// are not allowed, and updates the declaration state in `context`.
+auto CheckAccessModifiersOnDecl(Context& context, Lex::TokenKind decl_kind)
```

Comment:
You might consider renaming this file, e.g. to just "modifiers"

Reply:
Done.